samples-final/AspNetCore-LUIS-Bot/LuisBot.cs

```diff
@@ -0,0 +1,248 @@
+                await dc.Prompt("ShowReminderPrompt", "Select the reminder to show: ", new ChoicePromptOptions() { Choices = choices });
+            }
+        }
+
+        private async Task ConfirmShow(DialogContext dc, object args, SkipStepFunction next)
+        {
+            var userContext = dc.Context.GetUserState<UserState>();
+            if (args is Prompts.ChoiceResult choice)
+            {
+                var reminder = userContext.Reminders.FirstOrDefault(x => x.Title == choice.Value.Value);
+                await dc.Context.SendActivity($"Reminder: {reminder.Title} - ({reminder.Date})");
+            }
+            await dc.End();
+        }
+
+        private T GetEntity<T>(RecognizerResult luisResult, string entityKey)
+        {
+            var data = luisResult.Entities as IDictionary<string, JToken>;
+            if (data.TryGetValue(entityKey, out JToken value))
+            {
+                return value.First.Value<T>();
+            }
+            return default(T);
+        }
+
+        public async Task OnTurn(ITurnContext context)
+        {
+            if (context.Activity.Type == ActivityTypes.ConversationUpdate)
+            {
+                await context.SendActivity("Hi! I'm a simple reminder bot. I can help add reminders, delete and show them.");
+            }
+            else if (context.Activity.Type == ActivityTypes.Message)
+            {
+                var userState = context.GetUserState<UserState>();
+                if (userState.Reminders == null)
+                {
+                    userState.Reminders = new List<Reminder>();
+                }
+
+                var state = context.GetConversationState<Dictionary<string, object>>();
+                var dialogContext = dialogs.CreateContext(context, state);
+
+                var utterance = context.Activity.Text.ToLowerInvariant();
+                if (utterance == "cancel")
+                {
+                    if (dialogContext.Instance != null)
+                    {
+                        await context.SendActivity("Ok... Cancelled");
+                        dialogContext.EndAll();
+                    }
+                    else
+                    {
+                        await context.SendActivity("Nothing to cancel.");
+                    }
+                }
+                
+                if (!context.Responded)
+                {
+                    await dialogContext.Continue();
+
+                    if (!context.Responded)
+                    {
+                        var luisResult = context.Services.Get<RecognizerResult>(LuisRecognizerMiddleware.LuisRecognizerResultKey);
+                        var (intent, score) = luisResult.GetTopScoringIntent();
+                        var intentResult = score > LUIS_INTENT_THRESHOLD ? intent : "Fallback";
+
+                        await dialogContext.Begin(intent, luisResult);
```

Comment:
Unlike the Node.js SDK, a LUIS intent with a "." in the name, like "Calendar.Add", is coming back with the "." replaced with an underscore in the RecognizerResult. So this means invoking "Calendar.Add" leads to an exception when a dialog with an ad of 'Calendar.Add' isn't found.

Reply:
We updated the NuGet packages and modified the `DialogId`s